static/js/components/google_map.tsx

```diff
@@ -0,0 +1,257 @@
+  const marker = new google.maps.Marker({
+    position: location,
+  });
+  marker.setMap(map);
+}
+
+export class GoogleMap extends React.Component<
+  GoogleMapPropType,
+  GoogleMapStateType
+> {
+  div: React.RefObject<HTMLDivElement>;
+
+  constructor(props: GoogleMapPropType) {
+    super(props);
+    this.div = React.createRef();
+    this.state = {
+      markerLocation: { lat: null, lng: null },
+      mapInfo: {
+        up: null,
+        down: null,
+        left: null,
+        right: null,
+        coordinateSequenceSet: [],
+      },
+    };
+  }
+
+  render(): JSX.Element {
+    return <div className="map-container" ref={this.div}></div>;
+  }
+
+  componentDidMount(): void {
+    this.fetchData();
+  }
+
+  componentDidUpdate(): void {
+    // Initialize map
+    const map = initMap(this.div.current);
+
+    // plot KML Coordinates
+    if (Object.values(this.state.mapInfo).every((val) => val !== null)) {
+      drawKmlCoordinates(this.state, map);
+    }
+    // plot marker
+    else if (this.state.markerLocation.lat && this.state.markerLocation.lng) {
+      drawMarker(this.state.markerLocation, map);
+    }
+  }
+
+  private fetchData(): void {
+    // Skip API calls if we can get lat/long from DCID itself
+    if (
+      this.props.dcid.startsWith("latLong/") &&
+      this.props.dcid.includes("_")
+    ) {
+      this.setState({
+        markerLocation: convertLatLongDcid(this.props.dcid),
+      });
+    } else {
+      // Get KML Coordinates for polygons
+      const polygonPromise = axios
+        .get(`/api/place/mapinfo/${this.props.dcid}`)
+        .then((resp) => resp.data);
+
+      // Get lat/long from properties
+      const latitudePromise = axios
+        .get(`/api/browser/propvals/latitude/${this.props.dcid}`)
+        .then((resp) => resp.data);
+      const longitudePromise = axios
+        .get(`/api/browser/propvals/longitude/${this.props.dcid}`)
+        .then((resp) => resp.data);
+
+      Promise.all([polygonPromise, latitudePromise, longitudePromise]).then(
+        ([mapInfo, latitudes, longitudes]) => {
+          // Update state with mapInfo
+          if (mapInfo && Object.keys(mapInfo).length > 0) {
+            this.setState({
+              mapInfo: mapInfo,
+            });
+          }
+
+          // Update state with lat/long if both are present
+          if (
+            latitudes.values &&
+            longitudes.values &&
+            !_.isEmpty(latitudes.values.out) &&
+            !_.isEmpty(longitudes.values.out)
+          ) {
+            const latValues = latitudes.values.out.map(
+              (latValue) => latValue.value
+            );
+            const longValues = longitudes.values.out.map(
+              (longValue) => longValue.value
+            );
+            //TODO (juliawu): Update logic to use highest precision lat/long if
+            //                multiple values are provided
+            const coordinates = {
+              lat: parseFloat(latValues[0]),
```

Comment:
don't need to create latValues and longValues constants - can just use `latitudes.values.out[0].value` and `longitudes.values.out[0].value`?

Reply:
Good catch! That's much cleaner. Updated.